util/compiler.cpp

```diff
@@ -61,4 +61,12 @@ uint64_t divide_up(uint64_t n, uint64_t amount) {
   return (n + amount - 1) / amount;
 }
 
+uint64_t gcd(uint64_t n, uint64_t m) {
```

Comment:
uhm, there are more standard algorithms for this, such as: https://en.wikipedia.org/wiki/Greatest_common_divisor#Euclid's_algorithm

Reply:
This implements the algorithm :) with additional process when n < m & slightly optimize when n % m is already zero.